pkg/asset/installconfig/ibmcloud/validation.go

```diff
@@ -414,6 +417,52 @@ func ValidatePreExistingPublicDNS(client API, ic *types.InstallConfig, metadata
 	return nil
 }
 
+// ValidateServiceEndpoints will validate a series of service endpoint overrides.
+func ValidateServiceEndpoints(ic *types.InstallConfig) error {
+	allErrs := field.ErrorList{}
+	serviceEndpointsPath := field.NewPath("platform").Child("ibmcloud").Child("serviceEndpoints")
+	// Verify services are valid for override and are not duplicated and that are in valid URI format and accessible.
+	overriddenServices := map[configv1.IBMCloudServiceName]bool{}
+	for id, service := range ic.Platform.IBMCloud.ServiceEndpoints {
+		// Check if we have a duplicate service (case is ignored)
+		if _, ok := overriddenServices[service.Name]; ok {
+			allErrs = append(allErrs, field.Duplicate(serviceEndpointsPath.Index(id).Child("name"), service.Name))
+			continue
+		}
+		// Add service to map to track for duplicates
+		overriddenServices[service.Name] = true
+
+		// Check that the provided service name is an expected override service
+		if _, ok := ibmcloud.IBMCloudServiceOverrides[service.Name]; !ok {
+			allErrs = append(allErrs, field.Invalid(serviceEndpointsPath.Index(id).Child("name"), service.Name, "not a supported override service"))
+		}
+
+		// Check if the service URL is valid
+		err := validateEndpoint(service.URL)
+		if err != nil {
+			allErrs = append(allErrs, field.Invalid(serviceEndpointsPath.Index(id).Child("url"), service.URL, err.Error()))
+		}
+	}
+
+	return allErrs.ToAggregate()
+}
+
+// validateEndpoint will validate an endpoint meets acceptable URI requirements.
+func validateEndpoint(endpoint string) error {
+	// Ignore local unit tests
+	if endpoint == "e2e.unittest.local" {
+		return nil
+	}
+	// NOTE(cjschaef): At this time we expect the endpoint to be an absolute URI (besides local unittests checked above)
+	_, err := url.Parse(endpoint)
+	if err != nil {
+		return err
+	}
+	// Verify the endpoint is accessible, the endpoint check should be safe, as we only wish to validate the server responds
+	_, err = http.Head(endpoint) //nolint:gosec
```

Comment:
It would be nice to add a comment explaining why it's ok to silence the linter
```suggestion
	_, err = http.Head(endpoint) //nolint:gosec // reason goes here
```

Reply:
Sure, I'll add.